src/aiida/cmdline/groups/verdi.py

```diff
@@ -79,6 +80,37 @@ def __init__(self, *args, **kwargs):
             self.obj = LazyVerdiObjAttributeDict(self)
 
 
+class VerdiCommand(click.Command):
+    """Custom command implementation to customize the logic of printing deprecation messages.
+
+    If a command is deprecated, the :class:`click.Command` adds a deprecation marker in the short help and the full
+    help text, and prints a deprecation warning when the command is invoked. The problem is that the deprecation warning
+    is printed after the prompting for parameters, which for interactive commands mean the deprecation warning comes too
+    late, when the user has already provided all prompts.
+
+    Here, the :meth:`click.Command.parse_args` method is overridden, which is called before the interactive options
+    start to prompt, such that the deprecation warning can be printed. The :meth:`click.Command.invoke` method is also
+    overridden in order to skip the printing of the deprecation message handled by ``click`` as that would result in
+    the deprecation message being printed twice.
+    """
+
+    def parse_args(self, ctx: click.Context, args: t.List[str]) -> t.List[str]:
```

Comment:
this should fix the readthedocs error
```suggestion
    def parse_args(self, ctx: click.Context, args: t.List[str]) -> t.List[str]:
        """Given a context and a list of arguments this creates the parser and parses the arguments, then modifies the
        context as necessary. This is automatically invoked by :meth:`click.Command.make_context`.
        """
```

Reply:
hm.. still not

Reply:
Ah, the docs somehow inherit the docstring from the base class if it is not overridden and the base class contains the error in the reference. Didn't know that.

Reply:
Hmm, is that the right approach though? I think we _should_ inherit from `Command` shouldn't we?

Reply:
ah wait, I see, you meant in the docstring. I thought you changed the baseclass in the implementation there for a second 😅

Reply:
Can confirm that that fixes it though, thanks!